app/src/main/java/org/simple/clinic/patient/recent/RecentPatientsView.kt

```diff
@@ -33,16 +51,22 @@ class RecentPatientsView(context: Context, attrs: AttributeSet) : FrameLayout(co
       adapter = groupAdapter
     }
 
-    Observable.mergeArray(screenCreates())
+    Observable.mergeArray(screenCreates(), adapterEvents)
         .compose(controller)
         .takeUntil(RxView.detaches(this))
         .observeOn(AndroidSchedulers.mainThread())
         .subscribe { uiChange -> uiChange(this) }
+
+    adapterEvents
+        .ofType<RecentPatientItemClicked>()
+        .takeUntil(RxView.detaches(this))
+        .subscribe { openPatientSummary(it.patientUuid) }
```

Comment:
Handling events should be done in controller

Reply:
The reason I am not using controller for this case is because there is no decision-making required. It is just one `UiEvent` leading to a `UiChange` directly.
@saket and I have already discussed this. We are doing something similar in a few more places where a back click would lead to a screen change directly.

Reply:
Alright, can you change `Observable.mergeArray(screenCreates(), adapterEvents)` back to `Observable.mergeArray(screenCreates())`?

Reply:
Done